notifications_utils/recipients.py

```diff
@@ -286,20 +286,15 @@ def is_optional_address_column(self, key):
         )
 
     @property
-    def required_recipient_column_headers(self):
-        return set(
-            Columns.make_key(recipient_column)
-            for recipient_column in self.recipient_column_headers
-            if not self.is_optional_address_column(recipient_column)
-        )
+    def count_of_required_recipient_columns(self):
+        return 3 if self.template_type == 'letter' else 1
 
     @property
     def has_recipient_columns(self):
-        # `self.column_headers_as_column_keys` is an `OrderedSet`. We need
-        # to cast it to a set because `OrderedSet` has misleading
-        # behaviour when doing comparisions with `Set` objects.
-        # See: https://github.com/simonpercivall/orderedset/issues/25
-        return self.required_recipient_column_headers <= set(self.column_headers_as_column_keys)
+        return len(
+            self.recipient_column_headers_as_column_keys
+            & self.column_headers_as_column_keys
```

Comment:
Nice. Optional - a small inline comment just to remind people that `&` in this case is set intersection to save them having to figure that out in the docs

Reply:
Added comment in aa2ecd2